src/Approximations/overapproximate.jl

```diff
@@ -1510,3 +1510,123 @@ function overapproximate(am::AbstractAffineMap{N, <:AbstractHyperrectangle{N}},
     radius_MX = abs.(M) * radius_hyperrectangle(X)
     return Hyperrectangle(center_MXv, radius_MX)
 end
+
+"""
+    overapproximate(X::LazySet{N}, ::Type{<:Zonotope},
+                    dir::AbstractDirections{N};
+                    solver=default_lp_solver(N)) where {N<:Real}
+
+Overapproximate a polytopic set with a zonotope of minimal total generator sum
+using only generators in the given directions.
+
+### Input
+
+- `X`        -- polytopic set
+- `Zonotope` -- type for dispatch
+- `dir`      -- directions used for the generators
+- `solver`   -- (optional, default: `default_lp_solver(N)`) the backend used to
+                solve the linear program
+
+### Output
+
+A zonotope that overapproximates `X` and uses at most the directions provided in
+`dir` (redundant directions will be ignored).
+Under this constraint, the zonotope has the minimal sum of generator vectors.
+
+### Notes
+
+The algorithm only requires one representative of each generator direction and
+their additive inverse (e.g. only one of `[1, 0]` and `[-1, 0]`) and assumes
+that the directions are normalized.
+We preprocess the directions in that respect.
+
+### Algorithm
+
+We solve a linear program parametric in the vertices ``v_j`` of `X` and the
+directions ``d_k`` in `dir` presented in [1, Section 4.2] (adapting the notation
+to the one used in this library).
+
+```math
+    \\min \\sum_{k=1}^l α_k \\
+    s.t. \\
+    c + \\sum_{k=1}^l b_{kj} * d_k = v_j \\quad \\forall j \\
+    -α_k ≤ b_{kj} ≤ α_k \\quad \\forall k, j \\
+    α_k ≥ 0 \\quad \\forall k
+```
+
+The resulting zonotope has center `c` and generators `α_k · d_k`.
+
+Note that the first type of side constraints is vector-based and that the
+nonnegativity constraints (last type) are not stated explicitly in [1].
+
+[1] Zonotopes as bounding volumes, L. J. Guibas et al, Proc. of Symposium on
+    Discrete Algorithms, pp. 803-812.
+"""
+# note: this set can generally be reduced
```

Comment:
do you mean that it can be reduced without (or with a small) error?

Reply:
Huh, I do not understand this comment and it does not make any sense at the top of the function. I will remove it :sweat_smile: